### ISSUE TITLE
Duplicate symbols from constants when linking

### ISSUE THREAD

**Issue by davisp** [Microsoft/ChakraCore#2574]
This is fairly easy to demonstrate. I believe that declaring these three symbols as constants in ChakraCommon.h is a mistake as you can then never include ChakraCommon.h in multiple compilation units. Then again I don't have the greatest familiarity with the minutiae of C standards.

a.c
```c
#include <ChakraCore.h>

void do_hello_world(void);

int
main(int argc, char* argv[])
{
    do_hello_world();
}
```

b.c
```c
#include <stdio.h>
#include <ChakraCore.h>

void
do_hello_world(void)
{
    printf("Hello, World!\n");
}
```

Link failure
```bash
gcc -o bug a.c b.c
duplicate symbol _JS_SOURCE_CONTEXT_NONE in:
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/a-a26b4b.o
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/b-8eb111.o
duplicate symbol _JS_INVALID_RUNTIME_HANDLE in:
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/a-a26b4b.o
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/b-8eb111.o
duplicate symbol _JS_INVALID_REFERENCE in:
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/a-a26b4b.o
    /var/folders/4p/51v2mtgj0rzg_66d5vr3mygh0000gn/T/b-8eb111.o
ld: 3 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by davisp** [comment]
Also, I don't have a PR for this cause I'm not sure on the best fix. If it were me I'd probably move those to `#define`s but I'm not sure if there's a valid reason to keep them as `const` declarations and I haven't the foggiest if its valid to have an `extern const` or not.

**Comment by obastemur** [comment]
@davisp this is a valid issue in case multiple `c` source files includes ChakraCommon.h header file. This is a platform independent issue and I will send a PR shortly to fix it. Thanks for details!

**Comment by dmunch** [comment]
Just stumbled over this one, happy to see it's already fixed! And I'm even more curious to see what @davisp might be cooking up for CouchDB here ;)

**Comment by obastemur** [comment]
👍  -> `And I'm even more curious to see what @davisp might be cooking up for CouchDB here ;)`

BTW; fix is merged into `release/2.0`

**Comment by dmunch** [comment]
Great, thanks! Btw., what's the story behind `release/2.0` and `master`, i.e. is there a 2.0 release coming up soon? Tried to figure that out from the roadmap but the only information I could find was concerning `release/2.0-pre`
